src/model/cloud-runner/providers/provider-loader.ts

```
@@ -0,0 +1,48 @@
+    'runTaskInWorkflow',
+    'garbageCollect',
+    'listResources',
+    'listWorkflow',
+    'watchWorkflow',
+  ];
+
+  for (const method of requiredMethods) {
+    if (typeof instance[method] !== 'function') {
+      throw new Error(
+        `Provider package '${providerName}' does not implement ProviderInterface. Missing method '${method}'.`,
+      );
```

ESLint

[error] `new Error()` is too unspecific for a type check. Use `new TypeError()` instead.

(unicorn/prefer-type-error)